app/src/main/java/com/github/steroidteam/todolist/view/ListSelectionFragment.java

```diff
@@ -1,151 +1,156 @@
-                                        .thenAccept(
-                                                todoList -> {
-                                                    todoLists.add(todoList);
-                                                    adapter.notifyDataSetChanged();
-                                                });
-                            }
-                        }));
+        ItemTouchHelper itemTouchHelper = new ItemTouchHelper(new TodoTouchHelper(this));
+        itemTouchHelper.attachToRecyclerView(recyclerView);
 
         return root;
     }
 
-    private void setListViewSettings(ListView listView) {
-        listView.setAdapter(adapter);
-        listView.setLongClickable(true);
-
-        listView.setOnItemLongClickListener((adapterView, view, i, l) -> true);
+    public TodoArrayListAdapter.TodoHolder.TodoCustomListener createCustomListener() {
+        return new TodoArrayListAdapter.TodoHolder.TodoCustomListener() {
+            @Override
+            public void onClickCustom(TodoArrayListAdapter.TodoHolder holder) {
+                Bundle bundle = new Bundle();
+                bundle.putSerializable(EXTRA_LIST_KEY, holder.getTodo().getId());
+                Navigation.findNavController(holder.itemView).navigate(R.id.nav_item_view, bundle);
+            }
+        };
     }
 
     public void createList(View view) {
-        TodoList todoList = new TodoList(getString(R.string.new_todolist_title));
-        database.putTodoList(todoList)
-                .thenAccept(
-                        filePath -> {
-                            todoLists.add(todoList);
-                            adapter.notifyDataSetChanged();
-                        });
+        Context context = new ContextThemeWrapper(getActivity(), R.style.Dialog);
+        MaterialAlertDialogBuilder builder = new MaterialAlertDialogBuilder(context);
+        builder.setTitle(getString(R.string.add_todo_suggestion));
+
+        LayoutInflater inflater = this.getLayoutInflater();
+        View user_input = inflater.inflate(R.layout.alert_dialog_input, null);
+
+        builder.setView(user_input);
+
+        builder.setPositiveButton(
+                getString(R.string.add_todo),
+                (DialogInterface dialog, int which) -> {
+                    EditText titleInput = user_input.findViewById(R.id.alert_dialog_edit_text);
+                    String title = titleInput.getText().toString();
+                    if (title.length() > 0) viewModel.addTodo(title);
+                    titleInput.getText().clear();
+                    dialog.dismiss();
+                });
+        builder.setNegativeButton(
+                getString(R.string.cancel),
+                (DialogInterface dialog, int which) -> {
+                    dialog.dismiss();
+                });
+        Dialog dialog = builder.show();
+        dialog.getWindow().setGravity(0x00000035);
```

Comment:
What's this hex number? If we cannot replace it with something else (e.g. a `Gravity.*` constant or a value in `dimens.xml`), it'd be great to have a comment explaining it.